packages/palette_generator/lib/palette_generator.dart

```diff
@@ -69,8 +69,7 @@ class PaletteGenerator with Diagnosticable {
   ///
   /// The [paletteColors] argument must not be null.
   PaletteGenerator.fromColors(this.paletteColors, {this.targets})
-      : assert(paletteColors != null),
-        selectedSwatches = <PaletteTarget, PaletteColor>{} {
+      : selectedSwatches = <PaletteTarget, PaletteColor?>{} {
```

Comment:
Should the values really be nullable here?

Reply:
I think that potentially nullable values can occur there because values of this Map are assigned from a method that can return null. https://github.com/flutter/packages/blob/9f0a365d85c9e6b3561c3d227b226f66e28b257e/packages/palette_generator/lib/palette_generator.dart#L292

https://github.com/flutter/packages/blob/9f0a365d85c9e6b3561c3d227b226f66e28b257e/packages/palette_generator/lib/palette_generator.dart#L300

if this list is empty (not sure if can, but I can see a couple of .clear() calls) https://github.com/flutter/packages/blob/9f0a365d85c9e6b3561c3d227b226f66e28b257e/packages/palette_generator/lib/palette_generator.dart#L1154

https://github.com/flutter/packages/blob/9f0a365d85c9e6b3561c3d227b226f66e28b257e/packages/palette_generator/lib/palette_generator.dart#L312
then it skips this loop and returns `null`

Reply:
Do we actually want null values in the map though?

It seems like you're answering my questions by working forward from what the migration tool suggested. I'm asking about the *desired* behavior, which often requires adjusting the initial migration suggestions.

Reply:
Sure, makes sense. I didn't want to mess much with the logic (and didn't know that such refactoring is usually done on these migrations because as you said, it requires being aware and careful of code flow) because it is quite easy to miss some cases which handle these cases specifically.

Reply:
The point of the migration is to take code that was written for a non-null-safe world and convert it to being code that makes sense for a null-safe world. Making tons of things nullable during the migration undermines the goal of NNBD, and in many cases creates API issues that can only be fixed by doing new breaking changes in the future.

During the conversion to null safety is the best possible time to evaluate what should and should not allow null.

Reply:
This case with `PaletteColor`s ... seems like it is intented to return null to determine whether a color should be displayed with a Placeholder or not.

![image](https://user-images.githubusercontent.com/67197047/108111154-960f7c00-7094-11eb-8e6b-c0c7dbea2a19.png)

Would it make sense instead of making color nullable to create a new property like `bool isColorFound` and handle that in UI? However that would be breaking change https://github.com/flutter/packages/blob/9f0a365d85c9e6b3561c3d227b226f66e28b257e/packages/palette_generator/lib/palette_generator.dart#L257

Reply:
Please see https://github.com/flutter/packages/pull/287/commits/849516497291cb338a47a27c1f4cb1f987352ecb (did not change test - only for a demonstration to what I meant)

Reply:
I think you're misunderstanding my comment. I'm not suggesting you make the getters that look things up in the map non-nullable, I'm suggesting you make the *values in the map* non-nullable.

There's a difference between:
```
Map<String, Bar> bar;
Foo? => bar[baz];
```
and
```
Map<String, Bar?> bar;
Foo? => bar[baz];
```

The former is perfectly reasonable code. The latter is unlikely to be.

Reply:
Oh, I understand now. I'm dumb. I will make these changes and revert the breaking ones